airflow/api_connexion/schemas/dag_schema.py

```diff
@@ -149,6 +149,16 @@ def get_params(obj: DAG):
         return {k: v.dump() for k, v in params.items()}
 
 
+class DAGDetailSchemaWithTasksInfo(DAGDetailSchema):
+    """DAG with task ids details."""
+
+    tasks = fields.Method("get_tasks", dump_only=True)
+    @staticmethod
+    def get_tasks(obj: DAG) -> list[str]:
+        """Get DAG task ids."""
+        return list(obj.task_dict.keys())
```

Comment:
I wonder if it is possible/useful to add more details to the tasks (upstream list, downstream list, is setup? is teardown? ...)
I have no strong opinion on this, let's wait for other reviews before changing it

Reply:
@hussein-awala, regarding your question about adding more details to the tasks such as upstream lists, downstream lists, setup status, teardown status, and other functionalities, it's feasible and beneficial.

We can always add more details to the tasks as you mentioned. I have written flexible code to easily accommodate additional functionalities as part of the "DAGDetailSchemaWithTasksInfo" schema, as per future requirements. Task IDs play a crucial role, potentially saving numerous API calls, particularly when dealing with a large number of DAGs, if task IDs are included in the DAG details API info.

I understand your stance of waiting for other reviews before making any changes. Let's gather additional feedback to ensure alignment before proceeding with modifications.

Reply:
@ephraimbuddy @pierrejeambrun Can you please review the PR as well.